src/Runner.Listener/JobDispatcher.cs

```diff
@@ -767,6 +768,16 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
             }
         }
 
+        private void UpdateAgentNameIfNeeded(string agentName)
+        {
+            var configurationStore = HostContext.GetService<IConfigurationStore>();
```

Comment:
maybe just create a global `_configurationStore` in the `Initialize(IHostContext hostContext)` (we already have `_runnerSetting`), so we don't have to pull it out everytime.

Reply:
Right, we can rely on `_runnerSettings` and avoid another full `File.ReadAllText`. Also added `_configurationStore` as a prop 👍.